examples/events.py

```diff
@@ -0,0 +1,83 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+"""
+Example further illustrating the concept of 'events' which generalizes the idea of EOL. 
+
+'Events' is the term used to describe something to be predicted. 
+Generally in the PHM community these are referred to as End of Life (EOL). 
+However, they can be much more.
+
+In the implementation from the prog_models package, events can be anything that needs to be predicted. 
+Events can represent end of life (EOL), end of mission (EOM), warning thresholds, or any event of interest (EOI). 
+
+This example shows how a events can be used in your applications. 
+"""
+from prog_models.models import BatteryElectroChemEOD
+
+def run_example():
+    # Example: Warning thresholds
+    # In this example we will use the battery model
+    # We of course are interested in end of discharge, but for this example we
+    # have a requirement that says the battery must not fall below 5% SOC
+    # Additionally, we have two warning thresholds (yellow and red)
+
+    YELLOW_THRESH = 0.15
+    RED_THRESH = 0.1
+
+    # Step 1: Extend the battery model to define the additional events
+    class MyBatt(BatteryElectroChemEOD):
+        events = BatteryElectroChemEOD.events + ['EOD_warn_yellow', 'EOD_warn_red']
+
+        def event_state(self, state):
+            # Get event state from parent
+            event_state = super().event_state(state)
+
+            # Add yellow and red states by scaling EOD state
```

Comment:
This is a little hard to follow - is it scaling to make the EOD effectively 0% SOC?

Reply:
Ohhhh it's scaling between voltage levels and percentages, isn't it. This could probably use an extra comment, I got a bit lost here.